.circleci/config.yml

```diff
@@ -1,49 +1,69 @@
 version: 2
 jobs:
+  vault-token:
+    docker:
+      - image: circleci/buildpack-deps:curl
+    environment:
+      VAULT_ADDR: https://vault.afdevops.com
+    working_directory: /tmp
+    steps:
+      - run: mkdir -p workspace
+      - run: |
+          export VAULT_TOKEN=$(curl -s -X POST -d '{"project": "'${CIRCLE_PROJECT_REPONAME}'", "build_num": "'${CIRCLE_BUILD_NUM}'", "vcs_revision": "'${CIRCLE_SHA1}'"}' ${VAULT_ADDR}/v1/auth/circleci/login | jq -r '.auth.client_token')
+          export DOCKER_LOGIN=$(curl --header "X-Vault-Token: ${VAULT_TOKEN}" --connect-timeout 5 --max-time 10 -k -s ${VAULT_ADDR}/v1/secret/common/docker/docker_auth_key | jq -r '.data.value')
+          echo ${DOCKER_LOGIN} | base64 -d | awk -F: '{print $1}' > workspace/DOCKER_USER
+          echo ${DOCKER_LOGIN} | base64 -d | awk -F: '{print $2}' > workspace/DOCKER_PASS
+      - persist_to_workspace:
+          root: workspace
+          paths:
+            - DOCKER_USER
+            - DOCKER_PASS
   build:
     working_directory: /app
     environment:
       IMAGE_NAME: "${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME#docker-}:${CIRCLE_SHA1}"
     docker:
-      - image: docker:17.06.0-ce-git
+      - image: docker:17.09.0-ce-git
     steps:
       - checkout
       - setup_remote_docker
+      - attach_workspace:
+          at: /tmp/workspace
       - restore_cache:
           keys:
-            - v1-{{ .Branch }}
+            - v5-{{ .Branch }}-{{ .Revision }}
+            - v5-{{ .Branch }}-
+            - v5-
           paths:
             - /caches/app.tar
       - run:
           name: Load docker image layer cache
           command: |
-            set +o pipefail
-            docker load -i /caches/app.tar | true
+            docker load -i /caches/app.tar 2>&1 || true
       - run:
           name: Login to Dockerhub
           command: |
             set -u
-            docker login --username "${DOCKERHUB_USERNAME:-$CIRCLE_USERNAME}" --password "${DOCKERHUB_PASSWORD}"
+            cat /tmp/workspace/DOCKER_PASS | docker login -u $(cat /tmp/workspace/DOCKER_USER) --password-stdin
```

Comment:
@dmitriy-lukyanchikov are we still using developer account here or we have dedicated bot?

Reply:
We login with production user here